src/main/java/uk/gov/hmcts/darts/common/api/impl/CommonApiImpl.java

```diff
@@ -0,0 +1,45 @@
+package uk.gov.hmcts.darts.common.api.impl;
+
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.darts.common.api.CommonApi;
+import uk.gov.hmcts.darts.common.entity.Courthouse;
+import uk.gov.hmcts.darts.common.entity.Courtroom;
+import uk.gov.hmcts.darts.common.exception.DartsException;
+import uk.gov.hmcts.darts.common.repository.CommonCourthouseRepository;
+import uk.gov.hmcts.darts.common.repository.CourtroomRepository;
+
+import java.text.MessageFormat;
+import java.util.Locale;
+
+@Service
+@RequiredArgsConstructor
+@Slf4j
+public class CommonApiImpl implements CommonApi {
+
+    private final CourtroomRepository courtroomRepository;
+    private final CommonCourthouseRepository courthouseRepository;
+
+    @Override
+    public Courtroom retrieveOrCreateCourtroom(String courthouseName, String courtroomName) {
```

Comment:
I think this would need to be inside a Transaction

Reply:
moved to new @Transactional class